apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java

```diff
@@ -273,10 +275,12 @@ private Uni<OnboardingUtils.ProxyResource> getRegistryResource(Onboarding onboar
         };
     }
 
-    private Uni<Onboarding> persistOnboarding(Onboarding onboarding, List<UserRequest> userRequests, Product product) {
+    private Uni<Onboarding> persistOnboarding(Onboarding onboarding, List<UserRequest> userRequests, Product product, List<AggregateInstitutionRequest>aggregates) {
         /* I have to retrieve onboarding id for saving reference to pdv */
         return Panache.withTransaction(() -> Onboarding.persist(onboarding).replaceWith(onboarding)
                 .onItem().transformToUni(onboardingPersisted -> validationRole(userRequests)
+                        .onItem().transformToUni(ignore -> validateUserAggregatesRoles(aggregates))
+                        .onItem().transformToUni(ignore -> retrieveUserAggregatesResources(onboardingPersisted, product, aggregates))
```

Comment:
it might be better to name `retrieveAndSetUserAggregatesResources` because you are also setting the list into Onboarding object